packages/timeout/src/index.ts

```diff
@@ -82,30 +79,21 @@ export async function addTimeoutToPromise<T>(handler: () => Promise<T>, timeoutM
             if (!(e instanceof InternalTimeoutError)) {
                 throw e;
             }
-        } finally {
-            context.cancelTimeout.abort();
         }
     };
 
-    const timeout = async () => {
-        try {
-            await setTimeout(timeoutMillis, undefined, { signal: context.cancelTimeout.signal });
+    return new Promise((resolve, reject) => {
+        const timeout = setTimeout(() => {
             context.cancelTask.abort();
-        } catch (e) {
-            // ignore rejections (task finished and timeout promise has been cancelled)
-            return;
-        }
+            const error = errorMessage instanceof Error ? errorMessage : new TimeoutError(errorMessage);
+            reject(error);
+        }, timeoutMillis);
 
-        throw errorMessage instanceof Error
-            ? errorMessage
-            : new TimeoutError(errorMessage);
-    };
-
-    await new Promise((resolve, reject) => {
-        storage.run(context, () => {
-            Promise.race([timeout(), wrap()]).then(resolve, reject);
+        exports.storage.run(context, () => {
```

Comment:
```suggestion
        storage.run(context, () => {
```

Reply:
good catch, i was copy pasting from the compiled files from the repro and missed this one :D